### PR TITLE
fix(itemscontrol): Bind DisplayMemberPath target

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): fixes https://github.com/unoplatform/kahua-private/issues/7

Apply a binding rather than a one-time set to the property targeted by DisplayMemberPath. This ensures items update correctly if the property value should change.

## PR Type

What kind of change does this PR introduce?

- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs [hunk 1]
```diff
@@ -1046,18 +1046,20 @@ protected virtual void PrepareContainerForItemOverride(DependencyObject element,
...
 				{
-					return item;
+					container.SetValue(contentProperty, item);
 				}
 				else
 				{
-					// TODO: Cache the BindingPath
-					var b = new BindingPath(displayMemberPath, item) { DataContext = item };
-					return b.Value;
+					container.SetBinding(contentProperty, new Binding
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs:1058]: @jeromelaban Should this be checking if the binding already exists somehow?

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs:1058]: In general, it's best to validate if there's one with `GetBindingExpression`, particularly if we're reusing the container.

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs:1058]: That being said, we're clearing those `DataContext` and `Content` in `CleanUpContainer`, though we're clearing the bindings properly on `Clear()` so we may have duplicates there.

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs:1058]: Adjusted DependencyPropertyDetails to only keep a single binding, and dispose previous bindings.

File: src/Uno.UI.Tests/BinderTests/Given_Binder.TwoWay.cs [hunk 2]
```diff
@@ -162,5 +191,30 @@ public object ConvertBack(object value, Type targetType, object parameter, strin
...
+
+			public int MyValue
+			{
+				get => _value;
+				set
+				{
+					if (value != _value)
+					{
+						_value = value;
+						PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(MyValue)));
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/BinderTests/Given_Binder.TwoWay.cs:207]:
```suggestion
						PropertyChanged?.Invoke(this, new System.ComponentModel.PropertyChangedEventArgs(nameof(MyValue)));
```

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**